llvm/include/llvm/MC/MCPseudoProbe.h

```diff
@@ -213,6 +208,25 @@ class MCDecodedPseudoProbe : public MCPseudoProbeBase {
              bool ShowName) const;
 };
 
+// Address to pseudo probes map.
+class AddressProbesMap
+    : public std::vector<std::reference_wrapper<MCDecodedPseudoProbe>> {
+  auto getIt(uint64_t Addr) const {
+    auto CompareProbe = [](const MCDecodedPseudoProbe &Probe, uint64_t Addr) {
+      return Probe.getAddress() < Addr;
+    };
+    return llvm::lower_bound(*this, Addr, CompareProbe);
```

Comment:
How does this guarantee it's equivalent to `std::map<uint64_t, ....>` ? specifically, if the address doesn't exist in the map, `std::map` returns `end()` but using  `lower_bound`, it will points to the next element.

Reply:
Good catch. Initially I used equal_range in find(Addr) case, and lower_bound in find(From, To) case. Need to revert to equal_range.

Reply:
Actually, I remember now why I went with lower_bound in both cases: 
find(Addr) does the same as find(Addr, Addr+1), which in the case of missing Addr would return an empty range.

Reply:
Makes sense, thanks for the clarification. but `getIt(From), getIt(To))` introduces two vector search, can we just check if the return value's address is equal to the given address, if not, then return `end()`? this would save one search.